### PR TITLE
Add warning for v3 out of support. Fix incorrect build message.

### PR DESCRIPTION
**Pull request by jviau**
### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Fixes the function version printed when defaulting during build. This also adds a warning for functions v3 being out of support.

### CODE REVIEW

File: sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets [hunk 1]
```diff
@@ -55,7 +55,7 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
 
   <!-- Validating some expectations for the function app early on. -->
   <Target Name="_FunctionsPreBuild" BeforeTargets="BeforeBuild">
-    <Message Condition="'$(_AzureFunctionsNotSet)' == 'true'" Importance="high" Text="AzureFunctionsVersion not configured in the project. Setting AzureFunctionsVersion to v3"/>
+    <Message Condition="'$(_AzureFunctionsNotSet)' == 'true'" Importance="high" Text="AzureFunctionsVersion not configured in the project. Setting AzureFunctionsVersion to '$(AzureFunctionsVersion)'"/>
     <Error Condition="$(AzureFunctionsVersion.StartsWith('v1',StringComparison.OrdinalIgnoreCase))" Text="AzureFunctionsVersion is set to an incompatible version, Please set it to v4"/>
     <Error Condition="$(AzureFunctionsVersion.StartsWith('v2',StringComparison.OrdinalIgnoreCase))" Text="AzureFunctionsVersion is set to an incompatible version, Please set it to v3"/>
```
Comment (satvu) [inline sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets:60]: Can we also fix line 60 while we're here? Extended support for v3 ended in Dec 2022 so this is an outdated suggestion.

File: sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets [hunk 2]
```diff
@@ -45,6 +46,10 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
     <FunctionsGeneratedCodeNamespace Condition="'$(FunctionsGeneratedCodeNamespace)' == ''">$(RootNamespace.Replace("-", "_"))</FunctionsGeneratedCodeNamespace>
   </PropertyGroup>
 
+  <ItemGroup>
+    <_SupportedFunctionsVersion Include="v3;v4" />
```
Comment (mattchenderson) [inline sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets:50]: nit: use implies that this represents validity, not support.

File: sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets [hunk 3]
```diff
@@ -55,12 +60,17 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
...
-    <Error Condition="'$(_IsFunctionsSdkBuild)' == 'true'" Text="Microsoft.NET.Sdk.Functions package is meant to be used with in-proc function apps. Please remove the reference to this package in isolated function apps."/>
+    <PropertyGroup>
+      <_AzureFunctionsVersionStandardized>$(AzureFunctionsVersion.ToLowerInvariant())</_AzureFunctionsVersionStandardized>
+      <_ValidAzureFunctionsVersion Condition="'%(_SupportedFunctionsVersion.Identity)' == '$(_AzureFunctionsVersionStandardized)'">true</_ValidAzureFunctionsVersion>
+      <WarnOnDeprecatedFunctionsVersion Condition="'$(WarnOnDeprecatedFunctionsVersion)' == ''">true</WarnOnDeprecatedFunctionsVersion>
+    </PropertyGroup>
+
+    <Message Condition="'$(_AzureFunctionsNotSet)' == 'true'" Importance="normal" Text="AzureFunctionsVersion not configured in the project. Setting AzureFunctionsVersion to '$(_DefaultAzureFunctionsVersion)'" />
+    <Error Condition="'$(_ValidAzureFunctionsVersion)' != 'true'" Text="AzureFunctionsVersion is set to an unsupported value '$(AzureFunctionsVersion)'. Allowed values are: @(_SupportedFunctionsVersions, ', ')" />
```
Comment (mattchenderson) [inline sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets:70]: nit: "unsupported" feels off here. The check here is for validity in the context of isolated, hence why V3 is included in the collection. Nothing stated in the text is wrong, but I feel like stronger separation between these concepts is still valuable.

File: sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets [hunk 4]
```diff
@@ -55,12 +60,17 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
...
+    <Error Condition="'$(_ToolingSuffix)' == ''" Text="Invalid combination of TargetFramework and AzureFunctionsVersion is set." />
+    <Error Condition="'$(_IsFunctionsSdkBuild)' == 'true'" Text="Microsoft.NET.Sdk.Functions package is meant to be used with in-proc function apps. Please remove the reference to this package in isolated function apps." />
+    <Warning Condition="'$(WarnOnDeprecatedFunctionsVersion)' == 'true' AND '$(_AzureFunctionsVersionStandardized)' == 'v3'" Text="AzureFunctionsVersion v3 is out of support. Please consider upgrading to v4. https://aka.ms/azure-functions-retired-versions. Suppress this warning by adding '&lt;WarnOnDeprecatedFunctionsVersion&gt;false&lt;/WarnOnDeprecatedFunctionsVersion&gt;' to the project.'" />
```
Comment (mattchenderson) [inline sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets:73]: Should we bake in the v4 recommendation here? Would it make sense to use `_DefaultAzureFunctionsVersion` for futureproofing?

**Review by fabiocav — APPROVED** [review]

**Pull request merged**